modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/cache/ClientCachePartitionsRequest.java

```diff
@@ -187,15 +186,6 @@ private static boolean isApplicable(CacheConfiguration<?, ?> ccfg, boolean withC
         if (ccfg.getCacheMode() != CacheMode.PARTITIONED)
             return false;
 
-        IgnitePredicate<?> filter = ccfg.getNodeFilter();
```

Comment:
We can't remove this logic. Partition assignments are grouped into `ClientCachePartitionAwarenessGroup`, where caches have the same assignment. Caches with a node filter have different assignment, as explained below.

Reply:
Why the caches with different assigment are included in single ClientCachePartitionAwarenessGroup? Do we need to split into several separated ClientCachePartitionAwarenessGroup?

Reply:
Caches with same assignments are included in the same `ClientCachePartitionAwarenessGroup`. If there is a node filter, then we use separate `ClientCachePartitionAwarenessGroup` for such a cache.

I'm not sure if anything needs to be fixed or improved here.

Reply:
'If there is a node filter, then we use separate ClientCachePartitionAwarenessGroup for such a cache'

1) If I'm correct, we group caches with a node filter always in one `ClientCachePartitionAwarenessGroup` with null mapping. And never send the mapping to the client:
 ```
        ClientCachePartitionMapping mapping = null;

        if (isApplicable(cacheDesc.cacheConfiguration(), withCustomMappings))
            mapping = new ClientCachePartitionMapping(assignment);
```
That is why the partition awareness is skipped for such caches. Same for caches with custom affinity.

2) Another interesting issue is that grouping of partitions mapping doesn't seem to work. `equals()` and `hashCode()` of `ClientCachePartitionAwarenessGroup` do not function properly. `ClientCachePartitionAwarenessGroup` checks `ClientCachePartitionMapping` which checks its `HashMap<UUID, Set<Integer>> partitionMap`. The problem is that the set is actually `ImmutableIntSet`. It has no `equals()` and `hashCode()`. Same as the involved `BitSetIntSet`. 

Could you check the new tests?

Reply:
Approved, thank you.